### PR TITLE
only inject pod in namespace with annotation

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: Yang Keao <keao.yang@yahoo.com>

### What problem does this PR solve?

* Remove the "allowedNamespace" and "ignoredNamespace". 
* Only inject pods in the namespace with annotation `"chaos-mesh.org/inject": "enabled"`, so that it can be dynamically changed.

Annotation can be added to the namespace with:

```bash
kubectl annotate ns default chaos-mesh.org/inject=enabled
```

### Checklist

Tests

- [x] Unit test
- [ ] Manual test

Side effects

- [x] Breaking backward compatibility

Related changes

- [x] Need to update the documentation

### Does this PR introduce a user-facing change?

Yes.  :cry:

### CODE REVIEW

File: pkg/namespaceannoationfilter/filter.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package namespaceannoationfilter
```
Comment (AsterNighT) [inline pkg/namespaceannoationfilter/filter.go:14]: ughh, that's a very good long name indeed. Any idea making it more readable like dashes or making it shorter?

File: controllers/dnschaos/types.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ func (r *endpoint) Apply(ctx context.Context, req ctrl.Request, chaos v1alpha1.I
 		return err
 	}
 
-	pods, err := selector.SelectAndFilterPods(ctx, r.Client, r.Reader, &dnschaos.Spec, config.ControllerCfg.ClusterScoped, config.ControllerCfg.TargetNamespace, config.ControllerCfg.AllowedNamespaces, config.ControllerCfg.IgnoredNamespaces)
+	pods, err := selector.SelectAndFilterPods(ctx, r.Client, r.Reader, &dnschaos.Spec, config.ControllerCfg.ClusterScoped, config.ControllerCfg.TargetNamespace, config.ControllerCfg.EnableFilterNamespace)
```
Comment (AsterNighT) [inline controllers/dnschaos/types.go:64]: I'm not sure this compatibility break be merged into main or some other branch

Comment (YangKeao) [reply controllers/dnschaos/types.go:64]: `master` will be released as `1.2.0`, which will be safe for compatibility break.

@cwen0 Will we bring this feature with `1.2.0`?

File: pkg/namespaceannoationfilter/filter.go (hunk 1 above)
Comment (YangKeao) [reply pkg/namespaceannoationfilter/filter.go:14]: Yes. It's a really bad name. Do you have better idea?

Comment (AsterNighT) [reply pkg/namespaceannoationfilter/filter.go:14]: what about making it belong to something more general like a universal `podfilter`?

Comment (YangKeao) [reply pkg/namespaceannoationfilter/filter.go]: I moved it into the `selector`, (which contains a lot of `podfilter` and is really big :crying_cat_face: ).

**Review by AsterNighT — APPROVED** [review]

**Review by STRRL — DISMISSED** [review]
LGTM

**Comment by STRRL** [comment]
Should we merge this PR now?

**Comment by WangXiangUSTC** [comment]
> Should we merge this PR now?

Please wait for https://github.com/chaos-mesh/chaos-mesh/pull/1535, it needs to cherry-pick to release 1.1, and it has some conflicts with this pr

**Review by AsterNighT — APPROVED** [review]
LGTM, are we merging this since #1535 has been merged?

**Comment by STRRL** [comment]
Hi @YangKeao, here are some conflicts that block the merging, PTAL.

FYI @AsterNighT

**Comment by YangKeao** [comment]
@STRRL  @AsterNighT PTAL

**Review by STRRL — CHANGES_REQUESTED** [review]
I almost forget that we should update the docs: `helm/chaos-mesh/README.md`

**Review by AsterNighT — DISMISSED** [review]
LGTM

**Review by AsterNighT — APPROVED** [review]
LGTM

File: controllers/dnschaos/types.go (hunk 2 above)
Comment (cwen0) [reply controllers/dnschaos/types.go:64]: Yep

**Review by cwen0 — APPROVED** [review]
LGTM

**Comment by STRRL** [comment]
@YangKeao @cwen0, does this PR need to be cherry-picked to 1.1 and 1.2?

and the bot for cherry-pick seems to be broken, we might need do that manually.

**Comment by cwen0** [comment]
> @YangKeao @cwen0, does this PR need to be cherry-picked to 1.1 and 1.2?
> 
> and the bot for cherry-pick seems to be broken, we might need do that manually.

We can cherry-pick this pr to 1.2.  This change will affect some users, So we should not cherry-pick it to 1.1.

**Pull request merged**